File: frontend/src/types/goal.ts
export interface GoalUpdate {
  daily_calorie_target?: number | null;
  protein_target_g?: number | null;
  goal_weight_kg?: number | null;
}

export interface GoalOut {
  daily_calorie_target: number | null;
  protein_target_g: number | null;
  goal_weight_kg: number | null;
  updated_at: string;
}

File: backend/app/schemas/goal.py
from datetime import datetime
from pydantic import BaseModel


class GoalUpdate(BaseModel):
    daily_calorie_target: int | None = None
    protein_target_g: float | None = None
    goal_weight_kg: float | None = None


class GoalOut(BaseModel):
    daily_calorie_target: int | None
    protein_target_g: float | None
    goal_weight_kg: float | None
    updated_at: datetime

    class Config:
        from_attributes = True
